packages/common/container-definitions/src/runtime.ts

```diff
@@ -62,8 +62,10 @@ export enum AttachState {
 export interface IRuntime extends IDisposable {
 	/**
 	 * Notifies the runtime of a change in the connection state
+	 * @param canSendOps - true if the runtime is allowed to send ops
+	 * @param clientId - the id of the client that is connecting or disconnecting
```

Comment:
Nit
```suggestion
	 * @param clientId - The ID of the client that is connecting or disconnecting
```

Also, what does it mean for this to be undefined? Can we document that as well?

Reply:
`clientId` is only valid while a connection is established. It is provided by the service. Without an initial connection it must be `undefined`. Once there is a connection established, then we often keep the last `clientId` when disconnecting. Given the assertions in code to stay consistent with Audience's notion of `clientId`, I think it may not go back to `undefined`.